test/e2e/density.go

```diff
@@ -260,9 +267,59 @@ var _ = Describe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
+			uuid := string(util.NewUUID())
 			totalPods := itArg.podsPerMinion * minionCount
-			RCName = "my-hostname-density" + strconv.Itoa(totalPods) + "-" + string(util.NewUUID())
+			nameStr := strconv.Itoa(totalPods) + "-" + uuid
+			ns = "e2e-density" + nameStr
+			RCName = "my-hostname-density" + nameStr
+
+			// Create a listener for events
+			events := make([](*api.Event), 0)
+			_, controller := framework.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func() (runtime.Object, error) {
+						return c.Events(ns).List(labels.Everything(), fields.Everything())
+					},
+					WatchFunc: func(rv string) (watch.Interface, error) {
+						return c.Events(ns).Watch(labels.Everything(), fields.Everything(), rv)
+					},
+				},
+				&api.Event{},
+				time.Second*10,
+				framework.ResourceEventHandlerFuncs{
+					AddFunc: func(obj interface{}) {
+						events = append(events, obj.(*api.Event))
+					},
+				},
+			)
+			stop := make(chan struct{})
+			go controller.Run(stop)
+
+			// Start the replication controller
 			RunRC(c, RCName, ns, "gcr.io/google_containers/pause:go", totalPods)
+
+			By("Waiting for all events to be recorded")
+			last := -1
+			current := len(events)
+			timeout := 10 * time.Minute
+			for start := time.Now(); last < current && time.Since(start) < timeout; time.Sleep(10 * time.Second) {
+				last = current
+				current = len(events)
```

Comment:
I think it's a pretty harmless race, but you have the potential for concurrent reads and writes to events. Please send a follow-up PR that adds a lock.